src/main.rs

```diff
@@ -88,7 +89,13 @@ fn run() -> Result<ExitCode> {
                     ))
                 }
             })
-            .collect::<Result<Vec<_>>>()?,
+            .inspect(|res| {
+                if let Err(e) = res {
+                    print_error(format!("{}", e))
+                }
+            })
+            .filter_map(Result::ok)
+            .collect::<Vec<_>>(),
```

Comment:
we need to be careful here. With this change, `fd` will panic if NONE of the directories exists:
```
> fd . non-existing
[fd error]: Search path 'non-existing' is not a directory.
thread 'main' panicked at 'Error: Path vector can not be empty', src/walk.rs:51:26
```

Also, thank you for trying to keep the style of the code by using more iterator combinators. However, the whole thing doesn't seem very natural now: `map` wraps everything in `Ok(…)` or `Err(…)`. `inspect` then immediately unpacks this layer again and `filter_map` removes the `Ok(…)` wrapping layer.

I think the whole logic would probably be much simpler by using a for loop.

Reply:
I've reworked it as a for loop, and covered the missing case. See what you think.